src/javascript/app/pages/user/new_account/welcome_page.js

```diff
@@ -1,45 +1,59 @@
 const BinarySocket     = require('../../../base/socket');
 const Client           = require('../../../base/client');
 const getElementById   = require('../../../../_common/common_functions').getElementById;
-const localize         = require('../../../../_common/localize').localize;
 const Url              = require('../../../../_common/url');
-const createElement    = require('../../../../_common/utility').createElement;
 const showLoadingImage = require('../../../../_common/utility').showLoadingImage;
+const ClientBase       = require('../../../../_common/base/client_base');
+const State            = require('../../../../_common/storage').State;
 
 const WelcomePage = (() => {
+
+    let el_welcome_container,
+        is_virtual,
+        upgrade_info,
+        is_uk,
+        is_unwelcome_uk;
+
+    const init = () => {
+        upgrade_info          = ClientBase.getBasicUpgradeInfo();
+        is_virtual            = Client.get('is_virtual');
+        el_welcome_container  = getElementById('welcome_container');
+        is_uk                 = Boolean(Client.get('residence') === 'gb');
+        is_unwelcome_uk       = State.getResponse('get_account_status.status').some(status => status === 'unwelcome') && (/gb/.test(Client.get('residence')));
```

Comment:
same comments as the file above for these two.

Reply:
done did new function in client base